File: recipe-management-domain/src/main/java/fr/cta/recipe/management/domain/validation/Validation.java
package fr.cta.recipe.management.domain.validation;

import javax.validation.*;
import java.util.Set;

public interface Validation {

    String EXCEPTION_MESSAGE = "Bean validation error";

    Validator VALIDATOR = javax.validation.Validation.byDefaultProvider().configure()
        .buildValidatorFactory()
        .getValidator();

    static <T> void validate(T beanToValidate) {
        Set<ConstraintViolation<T>> constraintViolations = VALIDATOR.validate(beanToValidate);

        if (!constraintViolations.isEmpty()) {
            throw new ConstraintViolationException(EXCEPTION_MESSAGE + ":" + constraintViolations.toString(), constraintViolations);
        }
    }
}

File: recipe-management-domain/src/main/kotlin/fr/cta/recipe/management/domain/validation/Validation.kt
package fr.cta.recipe.management.domain.validation

import java.lang.IllegalStateException
import javax.validation.Validation
import javax.validation.Validator
import kotlin.reflect.KProperty0

object Validation {

    private val validator: Validator = Validation.byDefaultProvider().configure()
            .buildValidatorFactory()
            .validator

    fun validateNotBlank(propertyUnderValidation: KProperty0<String>) {
        if(propertyUnderValidation.get().isBlank())
            throw IllegalArgumentException("Property ${propertyUnderValidation.name} should not be blank")
    }

    fun validate(beanToValidate: Any) {
        val constraintsViolations = validator.validate(beanToValidate)

        if (constraintsViolations.isNotEmpty()) throw IllegalStateException(constraintsViolations.toString())
    }
}